src/Emit.hs

```diff
@@ -701,10 +701,11 @@ memberToDecl indent (memberName, memberType) =
 
 defStructToDeclaration :: Ty -> SymPath -> [XObj] -> String
 defStructToDeclaration structTy@(StructTy typeName typeVariables) path rest =
-  let indent' = indentAmount
+  let indent = indentAmount
 
       typedefCaseToMemberDecl :: XObj -> State EmitterState [()]
-      typedefCaseToMemberDecl (XObj (Arr members) _ _) = mapM (memberToDecl indent') (pairwise members)
+      typedefCaseToMemberDecl (XObj (Arr []) _ _) = sequence $ pure $ appendToSrc (addIndent indent ++ "char __dummy;\n")
```

Comment:
What is this fix about?

Reply:
Empty structs are illegal in C. There's at least one, the `(deftype Simple [])`
 in `examples/basics.carp`.

Reply:
lgtm--mind adding a comment that explains this? Could see it being an edge case we forget about ~N years down the line.